cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -158,6 +158,12 @@ protected BaseHiveExploreService(TransactionSystemClient txClient, DatasetFramew
     this.cConf = cConf;
     this.hConf = hConf;
     this.hiveConf = hiveConf;
+    String schedulerQueue = cConf.get(Constants.AppFabric.APP_SCHEDULER_QUEUE);
+    if (schedulerQueue != null && !schedulerQueue.isEmpty()) {
+      hConf.set(Constants.MapReduce.MAP_REDUCE_JOB_QUEUE_NAME, schedulerQueue);
```

Comment:
Every Explore execute call comes with a default namespace. Instead of determining namespace here, if we do it in `execute` method then we can figure out which namespace to use for each call.